.github/workflows/build.yml

```diff
@@ -56,7 +56,6 @@ jobs:
 
   build_for_macos:
     runs-on: macos-latest
-    if: github.event_name != 'pull_request' # To speed up the feedback
```

Comment:
In the past, the macos build took more than 30 min this is why we remove this jobs from PR event.
Last build took 16 min (~10 min for Linux) which is acceptable. So we can enable it back.